packages/compiler/core/checker.ts

```diff
@@ -3219,6 +3252,21 @@ export function createChecker(program: Program): Checker {
     return type;
   }
 
+  function resolveTypeReferenceString(reference: string): Type | undefined {
+    const parser = createParser(reference);
+    const node = parser.parseStandaloneReferenceExpression();
+    if (parser.parseDiagnostics.length > 0) {
+      program.reportDiagnostics(parser.parseDiagnostics);
```

Comment:
I actually think this should throw. It would be weird to get diagnostics in the emitter's typos in resolveTypeReference as user errors in the program.

Reply:
yeah, I was hesitating on this.